packages/vue-components/src/annotations/Annotate.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div :class="['annotate-image-wrapper', addClass]">
+    <img
+      ref="pic"
+      :src="src"
+      :alt="alt"
+      :width="computedWidth"
+      class="annotate-image"
+      :loading="computedLoadType"
+      @load.once="getWidth"
+    />
+    <div style="top: 0; left: 0; height: 0;">
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import { toNumber } from '../utils/utils';
+
+export default {
+  props: {
+    src: {
+      type: String,
+      default: null,
+    },
+    alt: {
+      type: String,
+      default: null,
+    },
+    height: {
+      type: String,
+      default: '',
+    },
+    width: {
+      type: String,
+      default: '',
+    },
+    eager: {
+      type: Boolean,
+      default: false,
+    },
+    addClass: {
+      type: String,
+      default: '',
+    },
+  },
+  computed: {
+    hasWidth() {
+      return this.width !== '';
+    },
+    hasHeight() {
+      return this.height !== '';
+    },
+    computedWidth() {
+      if (this.hasWidth) {
+        return this.width;
+      }
+      return this.widthFromHeight;
+    },
+    computedLoadType() {
```

Comment:
Actually this was taken from the Pic.vue class, so I wanted to keep it as similar as possible since there's a possibility that we can merge the two classes in the future.

Reply:
ah I see, in that case I agree that it's good to keep it as similar as possible